x-pack/plugins/apm/public/context/apm_service/apm_service_context.tsx

```diff
@@ -40,7 +43,10 @@ export function ApmServiceContextProvider({
     query,
   } = useApmParams('/services/:serviceName');
 
-  const { agentName } = useServiceAgentNameFetcher(serviceName);
+  const { agentMetadataDetails } = useServiceMetadataDetailsFetcher(
+    serviceName
+  );
+  const agentName = agentMetadataDetails?.service?.agent.name;
```

Comment:
Have you perf'd the difference between these two requests? The first I assume was very fast because it only returned the agent name. The second one I seem to recall is rather slow, which is why we only invoke it when the service icons are clicked (afair).

If this becomes slow a lot of UI elements, like conditional tabs, will render later making the UI look more janky.

Reply:
@sqren the `metadata` query is a bit slower than the `agent_name` query.

As you can see in the profiler: (Left: Agent_Name, Right: Metadata)
<img width="3199" alt="Screen Shot 2021-08-17 at 09 39 47" src="https://user-images.githubusercontent.com/55978943/129736300-4d1030ea-69a5-43d3-b973-51a96c8a92c9.png">
<img width="3200" alt="Screen Shot 2021-08-17 at 09 39 56" src="https://user-images.githubusercontent.com/55978943/129736306-74031be6-5ca1-473a-82e1-20560292dc29.png">

Now, if that's is going to harm the UX is hard to tell. I removed the `agent_name` API in order to simplify since the metadata already has the data that the agent_name would have to return in order to check what's the `service.runtime.name`. 

I could go to another direction, which could be:
- Return the service.runtime.name together with the agent name by calling the `agent_name` api.
- Create a new API to fetch only the service.runtime.name given a service name. In this case, I could only call this API if the service.name is ruby.